### DISCUSSION TITLE
Default and cached sorting method

### DISCUSSION THREAD

**Discussion by OrionTheGiant** [mealie-recipes/mealie#952] [Feature Request]
**Before submitting this feature request I have**
- [x] Reviewed the documentation and verified that my feature request isn't already a feature.
- [x] Reviewed the existing feature requests and my feature has not been requested.
- [x] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
It would be helpful to have a default sort selection on a per-user or global basis. Alternatively, the All Recipes page should cache the sort method when it's changed by the user so when navigating between pages, the sorting remains on what was previously selected.

**(Optional): Suggest A Solution**
A new prop needs to be created on the CardSection component and then run `sortRecipes()` with the value of the prop on one of the lifecycle hooks. The value retrieved from one of the two below methods can then be passed through the new CardSection prop in AllRecipes.vue
1. For the user or global setting: A new property called defaultSort can be added to the userSettings and/or globalSettings in the Vuex store. This does not address adding a place for changing the user or global setting within the UI
2. For a cached version, a similar property can be created in a different part of the Vuex store which should take precedence over the userSettings/globalSettings value.

**Additional Information**
- [x] If the feature is accepted I'm willing to submit a PR to provide this feature

**Comment by ArsVendy** [comment]
This would eliminate my only real annoyance with this otherwise completely excellent solution. Default sorting by Alphabetical should be a thing, I dislike having to select Alphabetical manually every time I change pages. Thanks for looking into this!

**Comment by epheterson** [comment]
I'm not even seeing any way to sort the "All Recipes" page at all. I do see that once I start a search, I can sort the results, though "All Recipes" page should have the same option, and totally agree that each user should be able to set a default sort (or the last-chosen sort should just stick for that user across all views).

**Reply by michael-genson** [reply]
The missing sort on the all recipes page is a temporary solution to a known issue: it's because of how the sorting is done (the sort doesn't actually work right). Once PR #1424 is done we can fix it.

This doesn't address the point of the discussion, but it addresses the missing sorting options on the all recipes page

**Reply by proffalken** [reply]
EDIT: I just discovered that I can sort on the "all recipes" page, that's fine for now, but being able to do it on the "welcome" page would be great as well!

Given that #1424 is now merged, does that we can hope to see the sorting options back on the front page in the near future? 
Ideally I'd like to sort by "highest rated" so that my family can rate all the dishes that are in there and I know which ones are actually going to be eaten!

**Reply by michael-genson** [reply]
Yup! Glad you found it.

The welcome page is really just a sort of "quick access" to find recent recipes. I'm not sure how sorting on that list would be different than sorting on the all recipes page.

**Comment by michael-genson** [comment]
This is now handled via browser storage and query params
